quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter.ts

```diff
@@ -63,32 +95,50 @@ export class SqlAutocompleter implements IAutocompleter {
    * @return {Promise<ICompleterItem[]>}
    */
   private async getQueryContextColumns(tables: TableInfo[]) {
-    const columnsNamesMemory: Set<string> = new Set();
-    const columnsWithPrefixMemory: Set<string> = new Set();
-    const tablesPromises: Promise<TableInfo>[] = [];
-
-    for (const table of tables) {
-      tablesPromises.push(this.extractTableColumns(table));
-    }
-    const extractedTables = await Promise.all(tablesPromises);
+    const extractedTables = await this.extractColumnsFromTable(tables);
+    const result: ComplexCompleterItem[] = [];
 
     for (const extractedTable of extractedTables) {
-      const { name, alias, columns, type } = extractedTable;
+      const { name, alias, columns } = extractedTable;
+      const includeTablePrefix = tables.length > 1;
       columns.forEach((column) => {
-        columnsNamesMemory.add(column);
+        if (typeof column === 'object' && typeof column.dataType === 'string') {
+          column.dataType = trinoToJs(column.dataType, 0);
+        }
 
+        let meta: string | undefined;
+        if (typeof column === 'object' && 'dataType' in column) {
+          const objectInTrino = 'row';
+          meta = typeof column.dataType === 'object' ? objectInTrino : column.dataType;
+        }
+
+        const getColumnDisplayName = (inputColumn: Column | string) => {
+          return inputColumn instanceof Object ? inputColumn.name : inputColumn;
+        };
+
+        let value: string;
         if (alias) {
-          columnsWithPrefixMemory.add(`${alias}.${column}`);
-        } else if (name && (tables.length > 1 || type === TableType.Nested)) {
-          columnsWithPrefixMemory.add(`${name}.${column}`);
+          value = `${alias}.${getColumnDisplayName(column)}`;
+        } else if (includeTablePrefix) {
+          value = `${name}.${getColumnDisplayName(column)}`;
+        } else {
+          value = getColumnDisplayName(column);
+        }
+
+        if (typeof column === 'string') {
+          result.push({ value: column, meta: 'column' });
+          if (alias || extractedTable.name) {
+            const prefix = alias || extractedTable.name;
+            result.push({ value: `${prefix}.${column}`, meta: 'column' });
+          }
+        } else {
+          result.push({ meta, value });
+          result.push({ meta, value: column instanceof Object ? column.name : column });
```

Comment:
fix `column instanceof Object ? column.name : column`

Reply:
i dont fully understand what you mean...